src/Integration/Binding/BindingWorkflow.cs

```diff
@@ -166,6 +171,27 @@ private ProgressStepDefinition[] CreateWorkflowSteps(IProgressController control
         #endregion
 
         #region Workflow steps
+        private void AbortWorkflow(IProgressController controller, CancellationToken token)
+        {
+            bool aborted = controller.TryAbort();
+            Debug.Assert(aborted || token.IsCancellationRequested, "Failed to abort the workflow");
+        }
+
+        internal /*for testing purposes*/ void PromptSaveSolutionIfDirty(IProgressController controller, CancellationToken token)
+        {
+            if (!VsShellUtils.SaveSolution(this.owner.ServiceProvider, silent: false))
+            {
+                VsShellUtils.WriteToGeneralOutputPane(this.owner.ServiceProvider, Strings.SolutionSaveCancelledBindAborted);
```

Comment:
Pity there's not an easy way to change the message that appears in the Team Explorer: that would be a better user experience.